src/pages/resultsView/ResultsViewPageStore.ts

```diff
@@ -1491,17 +1491,12 @@ export class ResultsViewPageStore {
 
     public readonly isHotspot = remoteData({
         await:()=>[
-            this.hotspotData,
-            this.indexedHotspotData
+            this.getOncoKbMutationAnnotationForOncoprint
         ],
         invoke:()=>{
             return Promise.resolve((mutation:Mutation)=>{
-                const indexedHotspotData = this.indexedHotspotData;
-                if (!indexedHotspotData.isComplete) {
-                    return false;
-                } else {
-                    return isHotspot(mutation, indexedHotspotData.result!.single);
-                }
+                const oncokbAnnotation = this.getOncoKbMutationAnnotationForOncoprint.result!(mutation);
```

Comment:
so the result is a function?

Reply:
this is a function that should be tracked as a mobx promise, ie not available before its dependencies are. If you have a better way to handle it, I'm open to it

Reply:
we've talked bout this before.  i would rather see dependencies "on the surface" and passed to pure functions because i think it's more flexible and explicit.  this caught my eye just because i hadn't seen a result be a function.  anyway, just an academic discussion.

Reply:
that would be nicer actually, I agree. lets handle that later?